misago/users/signals.py

```diff
@@ -1,11 +1,22 @@
 from django.dispatch import Signal, receiver
-
+from django.contrib.auth import get_user_model
+from misago.core.utils import ANONYMOUS_IP
 
 anonymize_user_content = Signal()
 delete_user_content = Signal()
 username_changed = Signal()
-
+anonymize_old_ips = Signal()
 
 @receiver(username_changed)
 def handle_name_change(sender, **kwargs):
     sender.user_renames.update(changed_by_username=sender.username)
+
+
+UserModel = get_user_model()
+MISAGO_IP_STORE_TIME = 33300
+
+@receiver(anonymize_old_ips)
+def handle_old_ips(sender, **kwargs):
+    for user in UserModel.objects.all():
```

Comment:
This will produce `MemoryError` on larger forum because `all()` queryset loads all results into memory before passing them to iterator. This query also fails to skip already-anonymized IP's or IPs that are too new to be anonymized which is something we don't want. ;)

Anonymization may happen completely in the database with query like this:

```python
anonymization_cutoff = timezone.now() - timedelta(days=settings.MISAGO_IP_STORE_TIME)
ip_is_too_new = Q(joined_on__lt=anonymization_cutoff)
ip_is_already_anonymized = Q(joined_from_ip=ANONYMOUS_IP)

queryset = UserModel.objects.exclude(ip_is_too_new | ip_is_already_anonymized)
queryset.update(joined_from_ip=ANONYMOUS_IP)
```

`timezone` can be imported via `from django.utils import timezone`, `timedelta` can be imported from `datetime`, and [`Q` object](https://docs.djangoproject.com/en/1.11/topics/db/queries/#complex-lookups-with-q-objects) from `django.db.models`